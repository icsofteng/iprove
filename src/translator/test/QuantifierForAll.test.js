const {translate} = require('../z3')
const {translate: translate_latex} = require('../latex')

const test_constants = [{ value: 'P', varType: 'Bool' }, { value: 'Q', varType: 'Bool' }]
const test_rules = [
  {
    type: 'universal_quantifier',
    symbol: 'forall',
    variables: [{ type: "variable", value: "x" , varType:'Type'}],
    value: {
      type: 'binary',
      symbol: 'implies',
      lhs: {
        type: 'literal',
        value: 'P'
      },
      rhs: {
        type: 'literal',
        value: 'Q'
      }
    }
  }
]

test('Quantifier Forall test', () => {
  expect(translate(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})

test('Quantifier Forall test latex', () => {
  expect(translate_latex(test_rules, test_constants, [], [], [])).toMatchSnapshot()
})
