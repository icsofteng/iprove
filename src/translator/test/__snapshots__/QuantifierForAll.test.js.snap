// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantifier Forall test 1`] = `
"(declare-const P Bool)
(declare-const Q Bool)
(assert (not (forall ((x Type))(=> P Q))))
(check-sat)"
`;

exports[`Quantifier Forall test latex 1`] = `
Array [
  "\\\\forall x:Type.P \\\\Longrightarrow Q",
]
`;
